openlibrary/plugins/importapi/code.py

```diff
@@ -459,7 +458,12 @@ def login(self, authstring):
         if not authstring:
             return
         authstring = authstring.replace("Basic ","")
-        username, password = base64.decodestring(authstring).split(':')
+        try:
+            authString = base64.decodebytes(bytes(authstring, 'utf-8'))
+            authString = authString.encode('utf-8')
```

Comment:
Shouldn't this be decode? On line 462, `authstring` starts as a base64 string and is encoded as utf-8 base64 bytes, and then decoded to normal bytes. At that point, `authString` (confusing naming to vary only by casing IMO) is bytes, so on line 463 you'd need to use [`bytes.decode`](https://docs.python.org/3/library/stdtypes.html#bytes.decode) to get a string, instead of [`str.encode`](https://docs.python.org/3/library/stdtypes.html#str.encode) to get bytes (which you can't `.split` on `':'` anyway). I think this code always raises "AttributeError: 'bytes' object has no attribute 'encode'", which is then handled by falling back to the deprecated method.

Reply:
Good eyes!!  And the trouble is that the _deprecated method_ was removed in Python 3.9.0!